### PR TITLE
feat: forward compatibility with v9 on `v-model` props/events

### PR DESCRIPTION
**Pull request by ShGKme**
### ☑️ Resolves

This PR adds **forward compatibility** with `next` (`v9`) on `v-model` props/events
- `modelValue` prop fallbacks to an old prop
- `update:modelValue` is emitted together with an old event
- `update:model-value` is emitted as well to use the new Vue 3 recommended event style in template compatible with Vue 2

Implementation details:
- Added `useModelMigration` to create migration proxy that:
  - Returns a new model prop if an old one is not defined
  - Triggers a new event together with an old one on set
  - Checks if either old or new prop is provided for required model
- Use `model` proxy instead of old prop to get value
- Assign to `model` instead of `$emit` an old event
- Define new `prop` and `event` as well as define/update `model` option
- Update docs:
  - Mark the old interfaces as deprecated in docs
  - Use `v-model` in docs examples

#### Docs

image

#### `required: true`

image

### 📃 Components

- Check components with `:checked` + `@update:checked`:
  - [x] NcActionCheckbox
  - [x] NcActionRadio
  - [x] NcCheckboxRadioSwitch
- Text input fields with `:value` + `@update:value`:
  - [x] NcActionTextEditable
  - [x] NcInputField
  - [x] NcPasswordField
  - [x] NcRichContenteditable
  - [x] NcTextArea
  - [x] NcTextField
- Selects (`:value` + `@input`):
  - [x] NcSelect
  - [x] NcSelectTags
  - [x] NcSettingsSelectGroup
- Pickers (`:value` + `@input`):
  - [x] NcColorPicker 
  - [x] NcDateTimePicker
  - [x] NcDateTimePickerNative
  - [x] NcTimezonePicker
- Special with `:value` + `@update:value`:
  - [x] NcActionInput
- Misc.:
  - [x] NcSettingsInputText (`:value` + `@update:value`)

### 🏁 Checklist

- [ ] ⛑️ Tests are included or are not applicable
- [x] 📘 Component documentation has been extended, updated or is not applicable
- [x] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

### CODE REVIEW

File: src/composables/useModelMigration.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { getCurrentInstance, computed } from 'vue'
+
+/**
+ * Create model proxy to new v9 model (modelValue + update:modelValue) with a fallback to old model
+ * @param {string} oldModelName - Name of model prop in nextcloud-vue v8
+ * @param {string} oldModelEvent - Event name of model event in nextcloud-vue v8
+ * @return {import('vue').WritableComputedRef} - model proxy
+ */
+export function useModelMigration(oldModelName, oldModelEvent) {
+    const vm = getCurrentInstance()!.proxy
```
Comment (susnux) [inline src/composables/useModelMigration.ts:15]: I am not sure here, I had issues with this in past (accessing the `proxy` inside of the setup function).
Not sure if it was something in my code but for me I had to move the `proxy` access into the computed functions

Comment (Antreesy) [reply src/composables/useModelMigration.ts:15]: We've been using similar approach for several major versions already in Talk for Vuex Store (useStore.js) in script setup components and composables, and so far didn't caught any errors with it.
Maybe some guard here with debug information would be safe (so if anyone encounters it, we can identify the reason)

Comment (ShGKme) [reply src/composables/useModelMigration.ts:15]: @susnux This is not possible. This function returns either `null` or `{ proxy: instance }`.

https://github.com/vuejs/vue/blob/main/src/v3/currentInstance.ts#L13

**Comment by ShGKme** [comment]
I understand this PR is quite huge. Maybe you can review at least some of them and check here?

- Components with `:checked` + `@update:checked`:
  - [ ] NcActionCheckbox
  - [ ] NcActionRadio
  - [ ] NcCheckboxRadioSwitch
- Text input fields with `:value` + `@update:value`:
  - [ ] NcActionTextEditable
  - [ ] NcInputField
  - [ ] NcPasswordField
  - [ ] NcRichContenteditable
  - [ ] NcTextArea
  - [ ] NcTextField
- Selects (`:value` + `@input`):
  - [ ] NcSelect
  - [ ] NcSelectTags
  - [ ] NcSettingsSelectGroup
- Pickers (`:value` + `@input`):
  - [ ] NcColorPicker 
  - [ ] NcDateTimePicker
  - [ ] NcDateTimePickerNative
  - [ ] NcTimezonePicker
- Special with `:value` + `@update:value`:
  - [ ] NcActionInput
- Misc.:
  - [ ] NcSettingsInputText (`:value` + `@update:value`)

**Review by Antreesy — APPROVED** [review]
Tested used in Talk components, do not expect the rest to break anything

**Comment by ShGKme** [comment]
🪄

File: src/components/NcSelectTags/NcSelectTags.vue [hunk 2]
```diff
@@ -116,11 +116,13 @@ export default {
...
 		:close-on-select="!multiple"
-		:value="passthru ? value : localValue"
+		:value="passthru ? model : localValue"
 		@search="searchString => search = searchString"
 		v-on="{
 			...$listeners,
-			input: passthru ? $listeners.input : handleInput,
+			input: passthru ? $listeners.input : noop,
+			'update:modelValue': passthru ? $listeners['update:modelValue'] : handleInput,
+			'update:model-value': passthru ? $listeners['update:model-value'] : noop,
```
Comment (skjnldsv) [inline src/components/NcSelectTags/NcSelectTags.vue:125]: image

Comment (skjnldsv) [reply src/components/NcSelectTags/NcSelectTags.vue:125]: @ShGKme

**Pull request merged**